src/domain.jl

```diff
@@ -0,0 +1,120 @@
+
+"""
+Construct subdomains by cell types using a temporary CPU backend.
+
+This implementation copies the cell type information to the CPU to simplify
+the construction of subdomains, then adapts the data back to the provided
+backend. This approach is used for performance and compatibility reasons.
+"""
+function Bcube.build_subdomains_by_celltypes(backend::BcubeBackendAcc, mesh, indices)
+    _ctypes = Bcube.cells(mesh)[indices]
+    indices_cpu = adapt(KernelAbstractions.get_backend([]), indices)
+    _ctypes_cpu = adapt(KernelAbstractions.get_backend([]), _ctypes)
+    ctypes_cpu = tuple(unique(_ctypes_cpu)...)
+    indice_by_ctypes_cpu = map(
```

Comment:
Is this map performed on the cpu or on the backend? Whyvnot leaving the whole function in Bcube and then just call adapt on it?